framework/src/ble_manager/ble_manager_autoconnect.c

```diff
@@ -0,0 +1,250 @@
+		evt = (ble_autocon_event)(buf[0]);
+		BLE_LOG_DEBUG("[BLEMGR] mqueue evt : %d\n", evt);
+
+		/* This is finishing process */
+		if (evt == BLE_AUTOCON_EVT_CANCEL || conn_stop == 1) {
+			conn_stop = 1;
+			if (state == BLE_AUTOCON_STATE_SCAN_STARTING) {
+				state = BLE_AUTOCON_STATE_SCAN_STOPPING;
+				ret = ble_client_stop_scan();
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn scan stop fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			} else if (state == BLE_AUTOCON_STATE_SCAN_STOPPING) {
+				if (evt == BLE_AUTOCON_EVT_SCAN_STOP) {
+					state = BLE_AUTOCON_STATE_DONE;
+				}
+			} else if (state == BLE_AUTOCON_STATE_CONNECTING) {
+				if (evt == BLE_AUTOCON_EVT_CONNECTED) {
+					state = BLE_AUTOCON_STATE_CONNECTED;
+					ctx->auto_connect = false;
+					break;
+				} else if (evt == BLE_AUTOCON_EVT_DISCONNECTED) {
+					state = BLE_AUTOCON_STATE_DONE;
+				}
+			} else {
+				state = BLE_AUTOCON_STATE_DONE;
+			}
+
+			if (state == BLE_AUTOCON_STATE_DONE) {
+				ctx->state = BLE_CLIENT_IDLE;
+				ctx->auto_connect = false;
+				break;
+			}
+
+		/* This is normal process */
+		} else if (state == BLE_AUTOCON_STATE_DISCONNECT) {
+			if (evt == BLE_AUTOCON_EVT_SCAN_START) {
+				state = BLE_AUTOCON_STATE_SCAN_STARTING;
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STARTING) {
+			if (evt == BLE_AUTOCON_EVT_MAC_SCANNED) {
+				state = BLE_AUTOCON_STATE_SCAN_STOPPING;
+				ret = ble_client_stop_scan();
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn scan stop fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STOPPING) {
+			if (evt == BLE_AUTOCON_EVT_SCAN_STOP) {
+				state = BLE_AUTOCON_STATE_CONNECTING;
+				ret = ble_client_reconnect(ctx);
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn reconnect fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_CONNECTING) {
+			if (evt == BLE_AUTOCON_EVT_CONNECTED) {
+				state = BLE_AUTOCON_STATE_CONNECTED;
+				break;
+			} else if (evt == BLE_AUTOCON_EVT_DISCONNECTED) {
+				state = BLE_AUTOCON_STATE_DISCONNECT;
+			}
+		}
+		BLE_LOG_DEBUG("[BLEMGR] auto conn state : %d\n", state);
+	}
+finish_auto:
+	mq_close(ctx->mqfd);
+	ctx->mqfd = 0;
+	mq_unlink(mq_name);
+	g_ctx = NULL;
+
+	BLE_LOG_INFO("[BLEMGR] Auto Connection Done\n");
+
+	return NULL;
+}
+
+ble_autocon_result_e ble_manager_autoconnect(ble_client_ctx *ctx)
+{
+	if (ctx->state != BLE_CLIENT_AUTOCONNECTING) {
+		return BLE_AUTOCON_INVALID_STATE;
+	}
+
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int status;
+	pthread_t g_thread = 0;
+
+	/* Initialize the attribute variable */
+	status = pthread_attr_init(&attr);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_attr_init failed, status=%d, errno=%d\n", status, errno);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* 1. set a priority */
+	sparam.sched_priority = 100;
```

Comment:
When you initialize the pthread attribute through `pthread_attr_init`,
it sets the priority as `100` and sets sched policy as `SCHED_RR`.
So I think you don't need to set these again.
You can only set the stack size what you want.

Reply:
@jeongchanKim 
Then, you mean only 'pthread_attr_setstacksize' is needed?

Reply:
If you change what I recommended, we can reduce so much codes :)

Reply:
I set pthread attr as default